tests/backup/backup_psa_test.go

```diff
@@ -1,30 +1,551 @@
+					err = CreateRestoreWithReplacePolicyWithValidation(defaultRestoreName, backupName, make(map[string]string), SourceClusterName, BackupOrgID, ctx, make(map[string]string), ReplacePolicyDelete, scheduledAppContexts)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creating default restore for backup with replace policy [%s]", defaultRestoreName))
+					restoreNames = append(restoreNames, defaultRestoreName)
+				}(backupName)
+			}
+			wg.Wait()
+		})
+
+		Step("Restore of applications with NS and StorageClass mapping with NS level PSA", func() {
+			log.InfoD("Restore of applications with NS and StorageClass mapping with NS level PSA")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			var wg sync.WaitGroup
+			for i, backupName := range backupNames {
+				wg.Add(1)
+				go func(backupName string) {
+					defer wg.Done()
+					defaultRestoreName := fmt.Sprintf("%s-%s-custom-ns-sc", RestoreNamePrefix, backupName)
+					pvcs, err := core.Instance().GetPersistentVolumeClaims(bkpNamespaces[i], labelSelectors)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Getting all PVCs from namespace [%s]. Total PVCs - %d", bkpNamespaces[i], len(pvcs.Items)))
+					singlePvc := pvcs.Items[0]
+					sourceScName, err := core.Instance().GetStorageClassForPVC(&singlePvc)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Getting SC from PVC - %s", singlePvc.GetName()))
+					storageClassMapping[sourceScName.Name] = scName
+					namespaceMapping[bkpNamespaces[i]] = bkpNamespaces[i] + "-restored"
+
+					err = CreateRestoreWithReplacePolicyWithValidation(defaultRestoreName, backupName, namespaceMapping, DestinationClusterName, BackupOrgID, ctx, storageClassMapping, ReplacePolicyDelete, scheduledAppContexts)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Create restore with NS and StorageClass mapping with NS level PSA"))
+					restoreNames = append(restoreNames, defaultRestoreName)
+				}(backupName)
+			}
+			wg.Wait()
+		})
+
+		Step("Restoring only PVC from backup", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			log.InfoD("Restoring backed up applications")
+			for i, backupName := range backupNames {
+				restoreName := fmt.Sprintf("%s-%s-%v", RestoreNamePrefix, backupName, time.Now().Unix())
+				namespaceMapping[bkpNamespaces[i]] = bkpNamespaces[i] + "-restored-only-pvc"
+				_, err := CreateCustomRestoreWithPVCs(restoreName, backupName, namespaceMapping, SourceClusterName, BackupOrgID, ctx, make(map[string]string), bkpNamespaces[i])
+				log.FailOnError(err, "Restoring of backup [%s] has failed with name [%s] in namespace [%s]", backupName, restoreName, fmt.Sprintf(bkpNamespaces[i]+"-restored-only-pvc"))
+				restoreNames = append(restoreNames, restoreName)
+			}
+		})
+
+		for _, psaLevel := range []string{"restricted", "baseline", "privileged"} {
+			Step(fmt.Sprintf("Setup PSA level to %s", psaLevel), func() {
```

Comment:
Can we add one more scenario where we deploy new application based on cluster level PSA and take backup and restore for this new app

Reply:
We have another case for that right?